src/lib/src/component/navigation/navigation.component.ts

```diff
@@ -159,4 +168,85 @@ export class NavigationComponent implements OnInit {
         || (this.uiState.selectedElement !== null && this.uiState.selectedElement.path.endsWith('.tcl'));
   }
 
+  onKeyUp(event: KeyboardEvent) {
+    console.log('KeyUp event received:\n' + event);
+    let element = this.uiState.selectedElement;
+    switch (event.key) {
+      case this.KEY_RIGHT: {
+        if (element !== null && element.type === ElementType.Folder) {
+          this.uiState.setExpanded(element.path, true);
+        }
+        break;
+      }
+      case this.KEY_LEFT: {
+        if (element !== null && element.type === ElementType.Folder) {
+          this.uiState.setExpanded(element.path, false);
+        }
+        break;
+      }
+      case this.KEY_DOWN: {
+        let successor = this.next(this.uiState.selectedElement);
+        if (successor != null) {
+          this.uiState.selectedElement = successor;
+          this.changeDetectorRef.detectChanges();
+        }
+        break;
+      }
+      case this.KEY_UP: {
+        let predecessor = this.previous(this.uiState.selectedElement);
+        if (predecessor != null) {
+          this.uiState.selectedElement = predecessor;
+          this.changeDetectorRef.detectChanges();
+        }
+        break;
+      }
+      case this.KEY_ENTER: {
+        if (element !== null && element.type === ElementType.File) {
+          this.messagingService.publish(events.NAVIGATION_OPEN, {
+            name: element.name,
+            path: element.path
+          });
+        }
+        break;
+      }
+    }
+  }
+
+  private next(element: WorkspaceElement): WorkspaceElement {
```

Comment:
should be something like nextVisible, since expanded flag etc. plays a significant role